webapp/src/shared/components/Timer/Timer.tsx

```diff
@@ -0,0 +1,106 @@
+import { useEffect, useState } from 'react';
+import { Text, TextVariant } from '../index';
+import './Timer.css';
+
+type TimerProps = {
+  shouldRun?: boolean;
+  textVariant?: TextVariant;
+  timeString?: string;
+  isBackwardsCount?: boolean;
+  onTimeOut?: () => void;
+};
+
+type clockType = {
+  timeInSeconds: number;
+  hours: number;
+  minutes: number;
+  seconds: number;
+};
+
+const secondsToClockType = (seconds: number): clockType => {
+  const hours = Math.floor(seconds / 3600);
+  const minutes = Math.floor(seconds / 60) - hours * 60;
+  const lseconds = Math.floor(seconds % 60);
+  return {
+    timeInSeconds: seconds,
+    hours,
+    minutes,
+    seconds: lseconds,
+  };
+};
+
+const parseTime = (input: string): clockType => {
+  let a = input.split(':'); // split it at the colons
+  if (a[0] && a[1] && a[2]) {
+    return {
+      timeInSeconds:
+        parseInt(a[0]) * 60 * 60 + parseInt(a[1]) * 60 + parseInt(a[2]),
+      hours: parseInt(a[0]),
+      minutes: parseInt(a[1]),
+      seconds: parseInt(a[2]),
+    };
+  }
+  throw new Error('Bad time format');
+};
+
+export default function Timer({
+  shouldRun = true,
+  textVariant = TextVariant.TITLE,
+  timeString = '00:01:00',
+  isBackwardsCount = true,
+  onTimeOut,
+}: TimerProps) {
+  const [startDate, _] = useState<clockType>(parseTime(timeString));
+  void _;
+  const [{ timeInSeconds, hours, minutes, seconds }, setTime] =
+    useState<clockType>(
+      isBackwardsCount
+        ? startDate
+        : {
+            timeInSeconds: 0,
+            hours: 0,
+            minutes: 0,
+            seconds: 0,
+          },
+    );
+  useEffect(() => {
+    if (shouldRun) {
+      setTimeout(() => {
```

Comment:
suggestion: We could add a `clearTimeout` here on the `useEffect` cleanup.

Reply:
Added (only in the paths where we actually want to clear it, although I'm not sure we do need it there. If we add it in the other paths, then the timer doesn't run).